### PR TITLE
DOCSP-38807: mdb 3.6 deprecation

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-scala/blob/master/REVIEWING.md)

**Note: this is part of a consolidation effort, so your review should
not address style guide issues or phrasing unless there are errors**

JIRA - <https://jira.mongodb.org/browse/DOCSP-38807>
Staging - https://preview-mongodbrustagir.gatsbyjs.io/scala/DOCSP-38807-server-3.6-eol/whats-new/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/whats-new.txt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+* :ref:`Version 5.1 <scala-version-5.1>`
+
+.. _scala-version-5.1:
+
+What's New in 5.1
+-----------------
+
+.. important:: MongoDB Server v3.6 End-of-Life
+
+   Support MongoDB Server v3.6 is deprecated in this
```
Comment (mongoKart) [inline source/whats-new.txt:32]: seems like a typo?

```suggestion
   MongoDB Server v3.6 is deprecated in this
```

File: source/whats-new.txt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+   release and will be removed in the v5.2 release of the driver.
+
+To learn about other changes introduced in v5.1, see the :driver:`Java Sync
+Driver What's New guide </java/sync/current/whats-new/>`.
```
Comment (mongoKart) [inline source/whats-new.txt:36]:
```suggestion
To learn about other changes introduced in v5.1, see the
:driver:`What's New guide </java/sync/current/whats-new/>`
for the Java Sync Driver.
```

**Review by mongoKart — APPROVED** [review]
LGTM + 2 small changes

File: source/whats-new.txt (hunk 1 above)
Comment (rustagir) [reply source/whats-new.txt:32]: oops should be support for

**Pull request merged**